### PR TITLE
fixed NPE message

### PR DESCRIPTION
**Pull request by palmanojkumar**

### CODE REVIEW

File: impl/src/main/java/org/ehcache/loaderwriter/writebehind/WriteBehindDecoratorLoaderWriterProviderFactory.java [hunk 1]
```diff
@@ -50,6 +50,9 @@ public void start(ServiceConfiguration<?> config, ServiceProvider serviceProvide
       
       @Override
       public <K, V> WriteBehindDecoratorLoaderWriter<K, V> createWriteBehindDecoratorLoaderWriter(CacheLoaderWriter<K, V> cacheLoaderWriter, WriteBehindConfiguration configuration) {
+        if (cacheLoaderWriter == null) {
+          throw new RuntimeException("WriteBehind requires non null CacheLoaderWriter.");
```
Comment (ljacomet) [inline impl/src/main/java/org/ehcache/loaderwriter/writebehind/WriteBehindDecoratorLoaderWriterProviderFactory.java]: A `NullPointerException` here is better than a simple runtime one - the message being the added information.

File: impl/src/test/java/org/ehcache/WriteBehindDecoratorLoaderWriterProviderFactoryTest.java [hunk 2]
```diff
@@ -81,6 +85,14 @@ public void testAddingWriteBehindConfigurationAtCacheLevel() {
     assertThat(serviceConfiguration, IsCollectionContaining.<ServiceConfiguration<?>>hasItem(instanceOf(WriteBehindConfiguration.class)));
     cacheManager.close();
   }
+  
+  @SuppressWarnings("unchecked")
+  @Test(expected=RuntimeException.class)
+  public void testWriteBehindWithoutCacheLoaderWriter() {
+    WriteBehindConfiguration writeBehindConfiguration = WriteBehindConfigurationBuilder.newWriteBehindConfiguration().build();
+    WriteBehindDecoratorLoaderWriterProviderFactory factory = new WriteBehindDecoratorLoaderWriterProviderFactory();
+    factory.create(mock(ServiceConfiguration.class), null).createWriteBehindDecoratorLoaderWriter(null, writeBehindConfiguration);
```
Comment (ljacomet) [inline impl/src/test/java/org/ehcache/WriteBehindDecoratorLoaderWriterProviderFactoryTest.java]: Most of the setup in this test is superfluous as calling into the last line with nulls only will result in the same exception.
However, doing it so would require to catch the exception and make sure we are getting the exact one we expect.

Comment (palmanojkumar) [inline impl/src/test/java/org/ehcache/WriteBehindDecoratorLoaderWriterProviderFactoryTest.java]: hmm. removed mock service configuration and doing specific exception test.

**Comment by ljacomet** [comment]
Content good, commit history + commit comments not good.

Can you please rebase in a single commit that references the issue?

**Comment by palmanojkumar** [comment]
Rebased all commits to single commit. test this please.

**Comment by ljacomet** [comment]
Format for referencing issues is #<issue> - see https://guides.github.com/features/issues/#notifications especially the references section.

**Pull request merged**